akka-stream/src/main/scala/akka/stream/scaladsl/One2OneBidiFlow.scala

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.scaladsl
+
+import akka.stream._
+
+import scala.util.control.NoStackTrace
+
+/**
+ * A generic BidiFlow which verifies that another flow produces exactly one output element per
```

Comment:
Yes, that was pretty much my solution evolution here.
Initially I started out with an AtomicInteger but quickly discovered that materializations become a problem.
Once again we'd need access to a value in one stage from another stage.
Maybe the new octopus stage would allow us to implement the complete bidi stage in a single entity? (Which could then create and use an AtomicInteger internally?)